### PR TITLE
Testing sign up endpoint

### PR DESCRIPTION
**Pull request by edilsonLon**
## Summary

- ### I fixed conflicts with the linter.
- ### I added a mandatory restriction for name, surname and password.
- ### I added script migration-test in package.json.
- ### I did the test for endpoint sign up

## Known Issues

- ### None

## Trello Card

- ### https://trello.com/c/mnhyDHJh/22-tests-de-sign-up

### CODE REVIEW

File: test/users/signup.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    const response = await request.post('/users').send(user);
+    expect(response.statusCode).toBe(201);
+    expect(response.body).toMatchObject(userResponse);
+  });
+  test('Create user when email exists must fail', async () => {
+    const { newUser: user } = UserData;
+    const userCreated = await factory.create('UserWithHash');
+    const response = await request.post('/users').send({ ...user, email: userCreated.email });
+    expect(response.statusCode).toBe(400);
+    expect(response.body).toMatchObject(UserErrors.emailExists);
```
Comment (martinremis) [inline test/users/signup.test.js]: I think UserErrors should be matched with the real errors, not with a factored object

File: test/factory/user_factory.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    password: 'abcdefghijkl'
+  },
+  userWithoutEmail: removeProperty('email')(userFake()),
+  userWithoutName: removeProperty('name')(userFake()),
+  userWithoutSurname: removeProperty('surname')(userFake()),
+  userWithoutPassword: removeProperty('password')(userFake()),
+  userWithoutParameters: {}
+};
+
+exports.UserErrors = {
```
Comment (martinremis) [inline test/factory/user_factory.js]: Why is this in PascalCase? We use PascalCase for clases or models (that works kind of like clases)

**Review by martinremis — CHANGES_REQUESTED** [review]

File: test/factory/user_factory.js (hunk 2 above)
Comment (edilsonLon) [reply test/factory/user_factory.js]: Sorry, my mistake, I replaced it with lowerCamelCase.

File: test/users/signup.test.js (hunk 1 above)
Comment (edilsonLon) [reply test/users/signup.test.js]: userErrors are real, but I replaced the strings with constants.🙂

**Review by martinremis — APPROVED** [review]

**Pull request merged**